jetbrains-ultimate/src-201+/software/aws/toolkits/jetbrains/datagrip/auth/SecretsManagerAuth.kt

```diff
@@ -61,8 +61,22 @@ class SecretsManagerAuth : DatabaseAuthProvider, CoroutineScope by ApplicationTh
             val project = connection.runConfiguration.project
             try {
                 val connectionSettings = getConfiguration(connection)
-                val credentials = getCredentials(connection.runConfiguration.project, connectionSettings)
-                DatabaseCredentialsAuthProvider.applyCredentials(connection, credentials, true)
+                val dbSecret = getDbSecret(connection.runConfiguration.project, connectionSettings)
+                if (connection.connectionPoint.additionalJdbcProperties[GET_URL_FROM_SECRET]?.toBoolean() == true) {
+                    dbSecret.host ?: throw IllegalArgumentException(message("datagrip.secretsmanager.validation.no_host", connectionSettings.secretId))
+                    dbSecret.port ?: throw IllegalArgumentException(message("datagrip.secretsmanager.validation.no_port", connectionSettings.secretId))
+                    // we have to rewrite the url which is pretty messy. The util is not better than using split (requires magic strings
+                    // to access the properties), so use split instead
+                    val db = connection.url.split("/").last()
+                    val jdbcUrlBeginning = connection.url.split("://").first()
+                    connection.url = "$jdbcUrlBeginning://${dbSecret.host}:${dbSecret.port}/$db"
```

Comment:
maybe something with the URI class? seems like it'd be more complicated than this though

Reply:
Yeah there's a JDBC url parser but I find it even more janky